src/components/timecard/edit-shift-hours/EditShiftHours.jsx

```diff
@@ -37,14 +37,44 @@ const EditShiftHours = ({
 
   const inputName = 'shift';
   const { setServiceError } = useApplicationContext();
-  const { timeEntries, setTimeEntries, timecardDate, setSummaryErrors } =
-    useTimecardContext();
+  const {
+    timeEntries,
+    setTimeEntries,
+    timecardDate,
+    summaryErrors,
+    setSummaryErrors,
+  } = useTimecardContext();
   const { userId } = useApplicationContext();
 
   const handleError = (errorFields) => {
     setSummaryErrors(errorFields);
   };
 
+  const handleServerValidationErrors = (error) => {
+    const summaryErrors = {};
+    let errorsHandled = true;
+
+    if (
+      error ==
+      ' has the following error(s): Time periods must not overlap with another time period'
+    ) {
+      summaryErrors['shift-start-time'] = {
+        message: 'Time periods must not overlap with another time period',
+      };
+      summaryErrors['shift-finish-time'] = {
+        message: '',
+      };
+    } else {
+      errorsHandled = false;
+    }
+
+    if (errorsHandled) {
+      setSummaryErrors(summaryErrors);
+      return true;
```

Comment:
What are these returned values being used for? Can we get rid of them?

Reply:
This is a tricky one because ideally this function would only be setting error messages, but in the situation where a backend error is thrown and we don't have a custom error message for it, we would want to show the service error page which is determined in `validateServiceErrors`. The only way we could get the information on whether all the errors have been handled is by returning a boolean